### ISSUE TITLE
Failed to parse input variable mapping expression when deploying process

### ISSUE THREAD

**Issue by deepthidevaki** [camunda-cloud/zeebe#7999]
**Describe the bug**

Cannot deploy a process with a valid expression. This was working fine with previous version of Zeebe.

```
Error: rpc error: code = InvalidArgument desc = Command 'CREATE' rejected with code 'INVALID_ARGUMENT': Expected to deploy new resources, but encountered the following errors:
'medic-week.bpmn': Failed to build variable mapping expression: failed to parse expression '{handoverTime: {   today: date and time(today(),time("13:30:00@Europe/Berlin")),   numberOfDays: for x in 1..7 return duration("P"+string(x)+"D"),   days: for d in numberOfDays return today + d,   fridays: days[day of week(item) = "Friday"],   nextFriday: fridays[1],   formatted: string(nextFriday) }.formatted}': Expected (end-of-input | continuation):1:302, found ".formatted"
```

**To Reproduce**

Deploy [process](https://raw.githubusercontent.com/zeebe-io/zeebe-medic-process/master/src/main/resources/medic-week.bpmn?token=AAPHVJS5JPBILND7N6XXAUTBOJX2E) 

Invalid expression (from the variable input mapping):
```
{
  handoverTime: {   
    today: date and time(today(), time("13:30:00@Europe/Berlin")),   
    numberOfDays: for x in 1..7 return duration("P"+string(x)+"D"),   
    days: for d in numberOfDays return today + d,   
    fridays: days[day of week(item) = "Friday"],   
    nextFriday: fridays[1],   
    formatted: string(nextFriday) 
  }.formatted
}
```

**Expected behavior**

No errors when deploying this process

**Environment:**
 - Zeebe Version: 1.2.0

**Comment by Zelldon** [comment]
@saig0 the medic process is because of this stuck now, has someone an idea ? :cry: 

```
java.lang.IllegalStateException: Failed to build variable mapping expression: failed to parse expression '{handoverTime: {   today: date and time(today(),time("13:30:00@Europe/Berlin")),   numberOfDays: for x in 1..7 return duration("P"+string(x)+"D"),   days: for d in numberOfDays return today + d,   fridays: days[day of week(item) = "Friday"],   nextFriday: fridays[1],   formatted: string(nextFriday) }.formatted}': Expected (end-of-input | continuation):1:302, found ".formatted
```


https://console.cloud.google.com/logs/viewer?interval=NO_LIMIT&authuser=0&organizationId=669107107215&project=camunda-cloud-240911&rif_reserved&minLogLevel=0&expandAll=false&timestamp=2021-11-02T07:33:16.474000000Z&customFacets=&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22camunda-cloud-240911%22%0Aresource.labels.location%3D%22europe-west1-d%22%0Aresource.labels.cluster_name%3D%22ultratest%22%0Aresource.labels.namespace_name%3D%225e3f5d89-61ab-4acb-a8e4-dbf7aaf07ee3-zeebe%22%0Aresource.labels.pod_name:%22zeebe-%22&scrollTimestamp=2021-11-02T07:33:09.438311000Z&pinnedLogId=0u0xmjhbxh8smsjc&pinnedLogTimestamp=2021-11-02T07:32:52.552707Z

**Comment by saig0** [comment]
As a workaround, we could flatten the expression. For example,

```
{
  today: date and time(today(), time("13:30:00@Europe/Berlin")),
  numberOfDays: for x in 1..7 return duration("P"+string(x)+"D"),
  days: for d in numberOfDays return today + d,   
  fridays: days[day of week(item) = "Friday"], 
  nextFriday: fridays[1],   
  handoverTime: string(nextFriday) 
}
```

**Comment by Zelldon** [comment]
For reference the instance is then blacklisted
```
E 2021-11-01T14:27:18.263167Z Expected to process event 'TypedEventImpl{metadata=RecordMetadata{recordType=COMMAND, intentValue=255, intent=COMPLETE_ELEMENT, requestStreamId=-2147483648, requestId=-1, protocolVersion=3, valueType=PROCESS_INSTANCE, rejectionType=NULL_VAL, rejectionReason=, brokerVersion=1.2.2}, value={"bpmnProcessId":"medic-week","version":13,"processDefinitionKey":2251799816591934,"processInstanceKey":2251799816852078,"elementId":"Activity_16jb7b8","flowScopeKey":2251799816852078,"bpmnElementType":"USER_TASK","parentProcessInstanceKey":2251799814940909,"parentElementInstanceKey":2251799816852075}}' without errors, but exception occurred with message 'Failed to build variable mapping expression: failed to parse expression '{handoverTime: {   today: date and time(today(),time("13:30:00@Europe/Berlin")),   numberOfDays: for x in 1..7 return duration("P"+string(x)+"D"),   days: for d in numberOfDays return today + d,   fridays: days[day of week(item) = "Friday"],   nextFriday: fridays[1],   formatted: string(nextFriday) }.formatted}': Expected (end-of-input | continuation):1:302, found ".formatted"'. 
W 2021-11-01T14:27:18.264979Z Blacklist process instance 2251799816852078, due to previous errors. 
W 2021-11-01T14:27:18.486922Z Blacklist process instance 2251799816852078, due to previous errors. 

```

 https://console.cloud.google.com/logs/viewer?interval=JUMP_TO_TIME&authuser=0&organizationId=669107107215&project=camunda-cloud-240911&rif_reserved&minLogLevel=0&expandAll=false&timestamp=2021-11-01T14:05:59.000000000Z&customFacets=&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22camunda-cloud-240911%22%0Aresource.labels.location%3D%22europe-west1-d%22%0Aresource.labels.cluster_name%3D%22ultratest%22%0Aresource.labels.namespace_name%3D%225e3f5d89-61ab-4acb-a8e4-dbf7aaf07ee3-zeebe%22%0Aresource.labels.pod_name:%22zeebe-%22%0AjsonPayload.context.loggerName!%3D%22io.camunda.zeebe.logstreams.snapshot%22%0AjsonPayload.context.loggerName!%3D%22io.camunda.zeebe.broker.exporter%22%0AjsonPayload.context.loggerName!%3D%22io.camunda.zeebe.journal.file.SegmentedJournal%22&scrollTimestamp=2021-11-01T14:27:18.264979000Z&dateRangeStart=2021-11-01T13:05:59.000Z&dateRangeEnd=2021-11-01T15:05:59.000Z&pinnedLogId=lqo6nb88e4hpjlyq&pinnedLogTimestamp=2021-11-01T14:27:18.264979Z

**Comment by saig0** [comment]
The issue is fixed by #8529. I verified the expression with the updated FEEL version in the REPL.
